plugins/janus_videoroom.c

```diff
@@ -4387,6 +4406,65 @@ static json_t *janus_videoroom_process_synchronous_request(janus_videoroom_sessi
 		json_object_set_new(response, "room", string_ids ? json_string(room_id_str) : json_integer(room_id));
 		json_object_set_new(response, "rtp_forwarders", list);
 		goto prepare_response;
+	} else if(!strcasecmp(request_text, "enable_recording")) {
+		JANUS_VALIDATE_JSON_OBJECT(root, record_parameters,
+			error_code, error_cause, TRUE,
+			JANUS_VIDEOROOM_ERROR_MISSING_ELEMENT, JANUS_VIDEOROOM_ERROR_INVALID_ELEMENT);
+		if(error_code != 0)
+			goto prepare_response;
+		json_t *record = json_object_get(root, "record");
+		int recording_active = json_is_true(record);
+		JANUS_LOG(LOG_VERB, "Enable Recording : %d \n", (recording_active ? 1 : 0));
+		/* Lookup room */
+		janus_mutex_lock(&rooms_mutex);
+		janus_videoroom *videoroom = NULL;
+		error_code = janus_videoroom_access_room(root, FALSE, TRUE, &videoroom, error_cause, sizeof(error_cause));
+		if(error_code != 0) {
+			JANUS_LOG(LOG_ERR, "Failed to access videoroom\n");
+			janus_mutex_unlock(&rooms_mutex);
+			goto prepare_response;
+		}
+		janus_mutex_lock(&videoroom->mutex);
+		janus_mutex_unlock(&rooms_mutex);
+		/* Set recording status */
+		videoroom->record = recording_active ? TRUE : FALSE;
```

Comment:
I feel this shouldn't update the value right away, but instead check what the value was before and do nothing if the new one is the same. Unless if, e.g., `videoroom->record` was `FALSE`, and there were individual recordings taking place, you want to still have them stopped anyway?

Reply:
I'll make this change.  Currently, it ensures the users recording state, which is interesting.  But I suppose an out-of-sync user is a sign of another issue.

Reply:
I wouldn't call it an issue: currently participants are able to individually change the state of their recording via `configure` messages, so a state where everyone except 1-2 people are being recorded can happen. In the future we may want to add an optional room property to prevent users from changing their recording state, and only allow it through the global feature you're adding now: this would ensure this desync cannot happen.

Reply:
Ah, true, great point.